src/arguments/utility.py

```diff
@@ -75,57 +77,102 @@ def get_ans(self, questions_list):
-                sys.exit()
-            json_ans_data = resp.json()
-
-            for data in json_ans_data["items"]:
-                output_content = [
-                    colored(
-                        "--------------------------------------------------------",
-                        'red'), data["body_markdown"],
-                    f"\U0001F517 Link to the answer:{data['link']}"
-                ]
-
+        # Create batch request to get details of all the questions
+        batch_ques_id = ""
+        for question_id in questions_list:
+            batch_ques_id += str(question_id) + ";"
+        try:
+            resp = requests.get(
+                f"{self.search_content_url}/2.2/questions/{batch_ques_id[:-1]}?order=desc&sort=votes&site=stackoverflow&filter=!--1nZwsgqvRX"
+            )
+        except:
+            SearchError("Search Failed", "Try connecting to the internet")
+            sys.exit()
+        json_ques_data = resp.json()
+        """
+        Store the received questions data into the following data format:
+        list(  list( question_title, question_link, question_id )  )
+        """
+        questions_data = [[item['title'], item['link'], item['question_id']] for item in json_ques_data["items"] ]
+        # Clear terminal
+        os.system('cls' if os.name == 'nt' else 'clear')
+
+        # cache array to store the requested answers. Format of storage { question_id:[answer_body, answer_link] }
+        downloaded_questions_cache = defaultdict(lambda: False)
+
+        # Stores the currently showing question index in questions_data
+        question_posx = 0
+
+        while True:
+            os.system('cls' if os.name == 'nt' else 'clear')
+            # Printing all the questions. The active question is printed GREEN.
+            console.rule('[bold blue] Relevant Questions', style="bold red")
+            for idx, question_data in enumerate(questions_data):
+                if question_posx == idx:
+                    console.print("[green]{}. {}  |  {}".format(idx+1, question_data[0], question_data[1]))
+                else:
+                    console.print("{}. {}  |  {}".format(idx+1, question_data[0], question_data[1]))
+            console.rule("[bold blue] Answer of question {}".format(question_posx+1), style="bold red")
+
+            # Gets the question_id of active question
+            current_question_id = questions_data[question_posx][2]
+
+            # Searches for the id in cache. If present then prints it
+            if(downloaded_questions_cache[current_question_id]):
+                output_content = downloaded_questions_cache[current_question_id]
                 for output_index, output_text in enumerate(output_content):
                     """
                     Loop through the output_text and print the element
-                    if it is the last one, the text[0] is printed
+                    if it the last one, the text[0] is printed
```

Comment:
keep it as `if it is the last one, the text[0] is printed` as it was earlier

Reply:
Okay